src/audio/eq_fir/eq_fir.c

```diff
@@ -643,26 +643,21 @@ static int eq_fir_copy(struct comp_dev *dev)
 {
 	struct comp_copy_limits cl;
 	struct comp_data *cd = comp_get_drvdata(dev);
-	int ret;
 	struct fir_state_32x16 *fir = cd->fir;
 	int nch = dev->params.channels;
 
 	tracev_comp("eq_fir_copy()");
 
 	/* Get source, sink, number of frames etc. to process. */
-	ret = comp_get_copy_limits(dev, &cl);
-	if (ret < 0) {
-		trace_eq_error("eq_fir_copy(): Failed comp_get_copy_limits()");
-		return ret;
-	}
+	comp_get_copy_limits(dev, &cl);
 
 	/* Check if number of frames to process if it is odd. The
 	 * optimized FIR function to process even number of frames
 	 * is lower load than generic version. In that case process
 	 * one frame with generic FIR and the rest with even frames
 	 * number FIR version.
 	 */
-	if (cl.frames & 0x1) {
+	if (cl.frames > 0 && cl.frames & 0x1) {
```

Comment:
No need to check > 0 since you are checking if LSB is set.

Reply:
Yep, you are right.

Reply:
Done now